dbms/src/Encryption/AESCTRCipherStream.cpp

```diff
@@ -189,6 +205,15 @@ BlockAccessCipherStreamPtr AESCTRCipherStream::createCipherStream(
     case EncryptionMethod::Aes256Ctr:
         cipher = EVP_aes_256_ctr();
         break;
+    case EncryptionMethod::SM4Ctr:
+#if OPENSSL_VERSION_NUMBER < 0x1010100fL || defined(OPENSSL_NO_SM4)
+        throw DB::TiFlashException("Unsupported encryption method: " + std::to_string(static_cast<int>(encryption_info_.method)),
+                                   Errors::Encryption::Internal);
+#else
+        // Openssl support SM4 after 1.1.1 release version.
+        cipher = EVP_sm4_ctr();
```

Comment:
Should we call `EVP_CIPHER_free` if exception throw after, and in `~ AESCTRCipherStream`?

https://github.com/openssl/openssl/blob/7f4c65749f7e9e687d5aa38e9b3fb548705511b9/demos/cipher/ariacbc.c#L163

Reply:
`EVP_CIPHER_free` should only be called on cipher which is get from `EVP_CIPHER_fetch` as shown in https://www.openssl.org/docs/manmaster/man3/EVP_CIPHER_fetch.html. 
And from the source code it is shown that `EVP_xxx_ctr` just return a pointer to an initialized static storage. So it don't need to be freed after use.

Reply:
We can add some comment about it

Reply:
Added in https://github.com/pingcap/tiflash/pull/5655/files/#diff-bcc8e47c4230c713d3018b3907a18e69cd02c7307396510e87618c41c2986033R184.